modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/util/DDMFormFieldTemplateContextContributorUtil.java

```diff
@@ -63,6 +67,51 @@ public static Map<String, Object> getLocaleMap(Locale defaultLocale) {
 		).build();
 	}
 
+	public static List<Map<String, Object>> getOptions(
+		DDMFormField ddmFormField,
+		ListTypeEntryLocalService listTypeEntryLocalService) {
+
+		List<Map<String, Object>> options = new ArrayList<>();
+
+		DDMFormFieldOptions ddmFormFieldOptions =
+			(DDMFormFieldOptions)ddmFormField.getProperty("options");
+
+		for (String optionValue : ddmFormFieldOptions.getOptionsValues()) {
+			if (optionValue == null) {
+				continue;
+			}
+
+			LocalizedValue localizedValue = ddmFormFieldOptions.getOptionLabels(
+				optionValue);
+
+			options.add(
+				HashMapBuilder.<String, Object>put(
+					"label",
+					localizedValue.getString(localizedValue.getDefaultLocale())
+				).put(
+					"labelMap",
+					() -> {
+						Map<Locale, String> labeMap = getListTypeEntryNameMap(
+							ddmFormField, optionValue,
+							listTypeEntryLocalService);
+
+						if (labeMap != null) {
+							return labeMap;
+						}
+
+						return localizedValue.getValues();
+					}
+				).put(
+					"reference",
+					ddmFormFieldOptions.getOptionReference(optionValue)
+				).put(
+					"value", optionValue
+				).build());
+		}
```

Comment:
this code is very similar to part of the code inside the method `SelectDDMFormFieldTemplateContext#getOptions()`. can we reuse this method there? maybe you will need to add an extra parameter to pass the specific locale to achieve the 2 scenarios.
what do you think?